jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -855,8 +855,16 @@ private static final MethodHandle resolveInvokeDynamic(long j9class, String name
 				throw new NullPointerException(Msg.getString("K05cd", classObject.toString(), bsmIndex)); //$NON-NLS-1$
 			}
 			Object[] staticArgs = new Object[BSM_OPTIONAL_ARGUMENTS_START_INDEX + bsmArgCount];
+			final MethodHandles.Lookup lookup = new MethodHandles.Lookup(classObject, false);
```

Comment:
The access check should be moved to immediately after the vmResolveFromMethodDescriptorString call so that access exceptions occur before any exceptions getting the BSM

Reply:
Will do.